### PR TITLE
Add Speculos ticker handling

### PR DESCRIPTION
**Pull request by xchapron-ledger**

### CODE REVIEW

File: src/ragger/backend/speculos.py [hunk 1]
```diff
@@ -28,6 +28,8 @@
 from ragger.utils import RAPDU, Crop
 from .interface import BackendInterface
 
+TICKER_DELAY = 0.1
```
Comment (lpascal-ledger) [inline src/ragger/backend/speculos.py:31]:
```suggestion
from speculos.mcu.seproxyhal import TICKER_DELAY
```

File: src/ragger/backend/speculos.py [hunk 2]
```diff
@@ -160,7 +169,7 @@ def left_click(self) -> None:
     def both_click(self) -> None:
         self._client.press_and_release("both")
 
-    def finger_touch(self, x: int = 0, y: int = 0, delay: float = 0.5) -> None:
+    def finger_touch(self, x: int = 0, y: int = 0, delay: float = 0.1) -> None:
```
Comment (lpascal-ledger) [inline src/ragger/backend/speculos.py:172]:
```suggestion
    def finger_touch(self, x: int = 0, y: int = 0, delay: float = TICKER_DELAY) -> None:
```
(not sure it makes total sense though)

File: src/ragger/navigator/navigator.py [hunk 3]
```diff
@@ -145,52 +144,20 @@ def _run_instruction(self,
...
+        if wait_for_screen_change:
+            if instruction.id == NavInsID.WAIT_FOR_SCREEN_CHANGE or \
+               instruction.id == NavInsID.WAIT_FOR_HOME_SCREEN or \
+               instruction.id == NavInsID.WAIT_FOR_TEXT_ON_SCREEN or \
+               instruction.id == NavInsID.WAIT_FOR_TEXT_NOT_ON_SCREEN:
+                # Function wait_for_screen_change() is already called during
+                # instruction callback execution above.
+                pass
+            else:
+                self._backend.wait_for_screen_change(timeout)
```
Comment (lpascal-ledger) [inline src/ragger/navigator/navigator.py:160]: I like this simplification.

File: src/ragger/navigator/navigator.py [hunk 4]
```diff
@@ -145,52 +144,20 @@ def _run_instruction(self,
...
-                    # instruction callback execution above.
-                    pass
-                else:
-                    self._backend.wait_for_screen_change(timeout)
+        # Call instruction callback
+        self._callbacks[instruction.id](*instruction.args, **instruction.kwargs)
+
+        # Wait for screen change unless explicitly specify otherwise
+        if wait_for_screen_change:
+            if instruction.id == NavInsID.WAIT_FOR_SCREEN_CHANGE or \
```
Comment (lpascal-ledger) [inline src/ragger/navigator/navigator.py:152]: Wouldn't
```python
if instruction.id in [NavInsID.WAIT_FOR_SCREEN_CHANGE, 
                      NavInsID.WAIT_FOR_HOME_SCREEN,
                      NavInsID.WAIT_FOR_TEXT_ON_SCREEN,
                      NavInsID.WAIT_FOR_TEXT_NOT_ON_SCREEN]:
```
... be clearer?

**Review by lpascal-ledger — APPROVED** [review]
Changelog update?

File: src/ragger/backend/speculos.py (hunk 2 above)
Comment (xchapron-ledger) [reply src/ragger/backend/speculos.py:171]: I prefer keeping the explicit 0.1 here.

**Pull request merged**